File: models/Gpt-3.5-turbo/py/code/mathematics/Exponentiation.py

MOD = 10**9 + 7

def modular_exponentiation(a, b):
    result = 1
    while b > 0:
        if b % 2 == 1:
            result = (result * a) % MOD
        a = (a * a) % MOD
        b //= 2
    return result

n = int(input())
for _ in range(n):
    a, b = map(int, input().split())
    print(modular_exponentiation(a, b))

File: models/Gpt-3.5-turbo/cpp/code/problems/mathematics/Exponentiation.cpp

#include <iostream>

int powerMod(int a, int b, int mod) {
    int result = 1;
    while (b > 0) {
        if (b & 1) {
            result = (result * 1LL * a) % mod;
        }
        a = (a * 1LL * a) % mod;
        b >>= 1;
    }
    return result;
}

int main() {
    int n;
    std::cin >> n;
    for (int i = 0; i < n; ++i) {
        int a, b;
        std::cin >> a >> b;
        std::cout << powerMod(a, b, 1000000007) << "\n";
    }
    return 0;
}
